src/Post_Command.php

```diff
@@ -748,4 +748,31 @@ private function read_from_file_or_stdin( $arg ) {
 		}
 		return file_get_contents( $readfile );
 	}
+
+	/**
+	 * Get category id if given name/slug
+	 *
+	 * @param string $arg Supplied argument.
+	 * @return array
+	 */
+	private function get_category_ids( $arg ) {
+		$categoires = explode( ',', $arg );
+
+		foreach ( $categoires as $post_category ) {
+			if ( trim( $post_category ) !== '' ) {
+				if ( is_numeric( $post_category ) && (int) $post_category ) {
+					$category_id = category_exists( (int) $post_category );
+				} else {
+					$category_id = category_exists( $post_category );
+				}
+				if ( ! $category_id ) {
+					$category_ids[] = $post_category;
+					WP_CLI::error( "No such post category '$post_category'." );
+				} else {
+					$category_ids[] = $category_id;
+				}
+			}
+		}
+		return $category_ids;
```

Comment:
For BC with previous behaviour, we need to do:
```
// If no category ids found, return exploded array for compat with previous WP-CLI versions.
return $category_ids ? $category_ids : $categories;
```
here.

The BC behaviour is weird and non-intuitive, in that `--post_category=' '` will set no categories, while `--post_category=0` will set the default category (see tests above), but probably we should keep this behaviour to be safe...??

Reply:
@gitlost  tests look really good.
so if valid categories are passed we will send the resolved ids 
if a non-existent one is passed then throw an error

Reply:
Thanks @thrijith !

>so if valid categories are passed we will send the resolved ids
if a non-existent one is passed then throw an error ?

I think so, giving an error is a BC break but makes more sense than just warning...